pkg/controller/jenkins/reconciler.go

```diff
@@ -117,32 +121,49 @@ func (r *JenkinsReconciler) Reconcile(request reconcile.Request) (reconcile.Resu
 	}
 	jenkinsSvc := newJenkinsService(instance, jenkinsInstanceName, jenkinsPort)                                  // jenkins service
 	jenkinsJNLPSvc := newJenkinsService(instance, jenkinsInstanceName+JenkinsJnlpServiceSuffix, jenkinsJNLPPort) // jenknis jnlp service
-	jenkinsPvc := newJenkinsPvc(instance, jenkinsInstanceName)                                                   // jenknis pvc
+	jenkinsRoute := newJenkinsRoute(instance, jenkinsSvc)
+	jenkinsPvc := newJenkinsPvc(instance, jenkinsInstanceName) // jenknis pvc
+	jenkinsServiceAccount := newJenkinsServiceAccount(instance, jenkinsInstanceName)
+	jenkinsRoleBinding := newJenkinsRoleBinding(instance, jenkinsInstanceName)
 
 	// Set Jenkins instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, dc, r.scheme); err != nil {
+	if err := controllerutil.SetControllerReference(instance, jenkinsDc, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// Check if this DC already exists
+	// Check if this DC already exists,
 	found := &appsv1.DeploymentConfig{}
 	namespacedName := types.NamespacedName{
 		Namespace: found.Namespace,
 		Name:      found.Name,
 	}
 
-	//TODO implements error checking
-	err = r.createResourceIfNotPresent(namespacedName, dc)
+	//TODO implement error checking for existence of the resource
+	// *or if there is an actual error* deal with it (<>_<>-)
+	err = r.createResourceIfNotPresent(namespacedName, jenkinsDc)
```

Comment:
Not related to this PR, but you might want to 
- read the DC if present
- populate relevant parts of it the usual way.
- execute an update/create.

( applies to all objects )

Reply:
Could the user do anything in the CR that might need the owned DC , service account to be updated?

Reply:
Have added a todo to update in the future